src/v3/src/transformer/layout/email/transformEmailAuthenticatorEnroll.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { NextStep } from '@okta/okta-auth-js';
+
+import {
+  ButtonElement,
+  ButtonType,
+  DescriptionElement,
+  IdxStepTransformer,
+  ReminderElement,
+  StepperButtonElement,
+  StepperLayout,
+  TitleElement,
+  UISchemaElement,
+  UISchemaLayout,
+  UISchemaLayoutType,
+} from '../../../types';
+import { getCurrentAuthenticator, loc } from '../../../util';
+import { getUIElementWithName } from '../../utils';
+
+export const transformEmailAuthenticatorEnroll: IdxStepTransformer = ({ transaction, formBag }) => {
+  const { nextStep = {} as NextStep, availableSteps } = transaction;
+  const { uischema } = formBag;
+  const authenticatorContextualData = getCurrentAuthenticator(transaction)?.value?.contextualData;
+  // @ts-ignore OKTA-551247 - useEmailMagicLink property missing from interface
+  const useEmailMagicLink = authenticatorContextualData?.useEmailMagicLink;
+
+  let reminderElement: ReminderElement | undefined;
+
+  const resendStep = availableSteps?.find(({ name }) => name?.endsWith('resend'));
+  if (resendStep) {
+    const { name } = resendStep;
+    reminderElement = {
+      type: 'Reminder',
+      options: {
+        content: loc('email.code.not.received', 'login'),
+        buttonText: loc('email.button.resend', 'login'),
+        step: name,
+        isActionStep: true,
+        actionParams: { resend: true },
+      },
+    };
+  }
+
+  const passcodeElement = getUIElementWithName(
+    'credentials.passcode',
+    uischema.elements as UISchemaElement[],
+  );
+  if (passcodeElement) {
+    passcodeElement.focus = true;
+  }
+
+  const subTitleElement: DescriptionElement = {
+    type: 'Description',
+    options: { content: '' },
+  };
+
+  if (typeof useEmailMagicLink === 'undefined') {
+    subTitleElement.options.content = loc('oie.email.enroll.subtitle', 'login');
+  } else {
```

Comment:
Ah, so in this case, when `useEmailMagicLink !== undefined` the Subtitle constructed still uses the `useEmailMagicLink` value. If its true it uses the `oie.email.verify.alternate.instructions` loc string, otherwise if its false it uses `oie.email.verify.alternate.verificationCode.instructions` loc string. 

See: https://github.com/okta/okta-signin-widget/pull/2939/files#diff-dcb7c5616f08957e8fe651c11c87e8506d109aabfd78adc4720c3d7ace2073deR75-R77

Reply:
Oh boy, it's a 3-state boolean value?

Reply:
:( unfortunately yes, this is why I ultimately separate Enroll vs verify :(

So its like:
```
if useEmailMagicLink === undefined : Show code entry form by default 
if useEmailMagicLink === false : Show slightly different subtitle text && show code entry form by default
if useEmailMagicLink === true : Show specific subtitle text && show stepper form
```

Reply:
Ok got it, thanks for explaining -- I think I am wary of the two ternaries below behaving based on two different types, one is switching on the `redactedEmailAddress` being truthy, whereas the second is switching on `useEmailMagicLink` being `true` (as boolean). Can we improve that a bit?